client/history/client.go

```diff
@@ -186,15 +186,13 @@ func (c *clientImpl) DescribeHistoryHost(
 func (c *clientImpl) RemoveTask(
 	ctx context.Context,
 	request *historyservice.RemoveTaskRequest,
-	opts ...grpc.CallOption) (*historyservice.RemoveTaskResponse, error) {
-	var err error
-	var client historyservice.HistoryServiceClient
-	if request.GetShardId() != 0 {
-		client, err = c.getClientForShardID(request.GetShardId())
-		if err != nil {
-			return nil, err
-		}
+	opts ...grpc.CallOption,
+) (*historyservice.RemoveTaskResponse, error) {
+	client, err := c.getClientForShardID(request.GetShardId())
```

Comment:
`getClientForShardID` will check if shardID <= 0 and return an invalid argument error.